src/creds/creds.c

```diff
@@ -832,13 +859,34 @@ static int parse_argv(int argc, char *argv[]) {
                         arg_tpm2_device = streq(optarg, "auto") ? NULL : optarg;
                         break;
 
-                case ARG_TPM2_PCRS:
+                case ARG_TPM2_PCRS: /* For fixed hash PCR policies only */
                         r = tpm2_parse_pcr_argument(optarg, &arg_tpm2_pcr_mask);
```

Comment:
Looks existing, but I was wondering what bank you select on? Does it default to the SHA256 bank. Perhaps in the future support tpm2-tools style with the bank selector would be a step towards making this algorithm agile.

Example: `sha1:3,4+sha256:all`

Reply:
The `systemd-creds` tool can be used to encrypt so called "credentials" that can be passed to system services, and are then decrypted only when those services are activated. This encryption/decryption combines a key we store in the file system with one we unseal via the TPM. Depending on the environment we might only use one of these keys. ("Credentials" are entirely independent from LUKS disk encryption btw, different concept)

Encryption/decryption is symmetric and supposed to happen on the system these credentials are stored and used. Thus, when we encrypt we pick the most suitable PCR bank. Specifically this means SHA256 if it is enabled in the TPM, and looks initialized. If SHA256 is disabled or doesn't look used (i.e. is all zeroes for the selected PCRs) we'll look at the SHA1 bank instead. If that one also doesn't look used we'll go for the SHA256 one after all, even if it doesn't look used. We won't consider the other banks if they exist.

We started doing this dance after getting reports from people who have systems that claim to support SHA256 banks but the firmware doesn't actually measure anything into them, and only measures to the SHA1 bank. 

The bank used is stored in the encrypted credential metadata. So when decrypting the automatic selection of the bank used is not repeated.